framework/include/utils/CompileTimeDerivatives.h

```diff
@@ -409,44 +631,64 @@ class CTIPow : public CTUnary<B>
       return pow<E - 1>(_arg) * E * _arg.template D<dtag>();
   }
 
-  using typename CTUnary<B>::O;
   using CTUnary<B>::_arg;
 };
 
+/**
+ * pow<exponent>(base) template for integer powers.
+ */
+template <int E, typename B>
+auto
+pow(const B & base)
+{
+  if constexpr (std::is_base_of<CTBase, B>::value)
+    return CTIPow<B, E>(base);
+  else
+    return CTIPow<CTValue<CTNoTag, B>, E>(makeValue(base));
+}
+
 /**
  * Macro for implementing a binary math operator overload that works with a mix of CT system
  * objects, C variables, and number literals.
  */
 #define CT_OPERATOR_BINARY(op, OP)                                                                 \
```

Comment:
It's nice to know that, even with C++17, I'm not the only one who can't figure out how to do this kind of metaprogramming cleanly without oodles of preprocessor macros.